neotest/main.d

```diff
@@ -159,6 +159,56 @@ class Get : DhtTest
     }
 }
 
+class Update : DhtTest
+{
+    import ocean.core.array.Mutation : copy;
+
+    override protected void go ( )
+    {
+        this.dht.neo.update("test".dup, 0, &this.notifier);
+        this.suspend();
+    }
+
+    private void notifier ( DhtClient.Neo.Update.Notification info,
+        Const!(DhtClient.Neo.Update.Args) args )
+    {
+        formatNotification(info, this.dht.msg_buf);
+        Stdout.formatln("{}", this.dht.msg_buf);
+
+        with ( info.Active ) final switch ( info.active )
+        {
+            case received:
+                auto received_record = info.received.value;
+                (*info.received.updated_value).copy(received_record);
+                (*info.received.updated_value) ~= [cast(ubyte)'X'];
+                break;
+
+            case conflict: // Another client updated the same record. Try again.
+                Stdout.formatln("Finished: conflict");
+                this.resume();
+                break;
+
+            case succeeded: // Updated successfully.
+            case no_record: // Record not in DHT. Use Put to write a new record.
+                Stdout.formatln("Finished: OK");
+                this.resume();
+                break;
```

Comment:
I'm curious if it's beneficial for the client to allow `Put` request here just by assigning to `info.received.value` (in case of `no_record`).

Reply:
I originally planned for it to work like that, but removed it for the sake of reducing complexity -- I just wanted to get the basic thing working first. It's something we can look into adding, down the line.

Reply:
Fair enough :+1: